libopenage/game_control.cpp

```diff
@@ -41,13 +41,31 @@ void CreateMode::on_exit() {}
 
 void CreateMode::render() {}
 
+ActionModeSignals::ActionModeSignals(ActionMode *action_mode)
+	:
+	action_mode(action_mode) {
+}
+
+void ActionModeSignals::on_action(const std::string &action_name) {
+	Engine &engine = Engine::get();
+	input::action_t action = engine.get_action_manager().get(action_name);
+	input::InputContext *top_ctxt = &engine.get_input_manager().get_top_context();
+	if (top_ctxt == this->action_mode || top_ctxt == &this->action_mode->building_context ||
+	    top_ctxt == &this->action_mode->build_menu_context || top_ctxt == &this->action_mode->build_menu_mil_context) {
+		using namespace input;
```

Comment:
I swear I stole this from somewhere else but I can't find it right now - I believe the issue was that it would be `input::action_arg_t action_arg{input::Event{input::event_class::ANY, 0, input::modset_t{}}`... which is a tad verbose. Maybe there should be a function under openage::input that creates an `action_arg_t` like this?